rockcraft/layers.py

```diff
@@ -54,6 +55,32 @@ def archive_layer(
             tar_file.add(filepath, arcname=arcname, recursive=False)
 
 
+def prune_prime_files(prime_dir: Path, files: Set[str], base_layer_dir: Path) -> None:
+    """Remove (prune) files in a prime directory if they exist in the base layer.
+
+    Given a set of filenames ``files``, this function will remove (prune) all those
+    filenames from prime dir ``prime_dir`` if the corresponding sub path exists in
+    the extracted base layer directory ``base_layer_dir`` with same contents and
+    permissions.
+
+    For example, "{prime_dir}/dir/subdir/file1" will be pruned if the matching file
+    "{base_layer_dir}/dir/subdir/file1" exists and has the same contents, owner,
+    group, and permission bits.
+    """
+    emit.debug(f"Pruning contents of {prime_dir} due to base layer {base_layer_dir}...")
```

Comment:
This sentence might need some love :) it's not "due to" but something like "that conflict with"

Reply:
In my opinion "that conflict with" is misleading: from craft-part's point-of-view, a conflict is when two files have the same name but different contents/permissions. In this case the pruning happens exactly when there is a filename match but _no_ conflict.
How about "Pruning primed files that already exist on base layer..."?